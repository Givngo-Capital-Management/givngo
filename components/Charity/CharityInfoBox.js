import React from 'react';
import { Image, Button } from 'semantic-ui-react';
import styles from './CharityInfoBox.module.scss';

export default function CharityInfoBox() {
  return (
    <div className={styles.box}>
      <Image
        className={styles.image}
        src="http://via.placeholder.com/48x48"
        circular
      />
      <div className={styles.info}>
        <div className="charity-name">Charity Name</div>
        <span>80k followers</span>
      </div>
      <Button className={styles.follow} color="orange">
        Follow
      </Button>
      <div className={styles.description}>
        <p>
          Lorem Ipsum is simply dummy text of the printing and typesetting
          industry. Lorem Ipsum has been the industry&apos;s standard dummy text
          ever since the 1500s, when an unknown printer took a galley of type
          and scrambled it to make a type specimen book. It has survived not
          only five centuries, but also the leap into electronic typesetting,
          remaining essentially unchanged. It was popularised in the 1960s with
          the release of Letraset sheets containing Lorem Ipsum passages, and
          more recently with desktop publishing software like Aldus PageMaker
          including versions of Lorem Ipsum.
        </p>
      </div>
    </div>
  );
}
