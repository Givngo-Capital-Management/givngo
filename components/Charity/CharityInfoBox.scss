.charity-info-box {
  display: grid;
  grid: auto auto / 58px auto max-content;
  align-items: center;
  grid-row-gap: 16px;

  .charity-image {
    grid-row: 1 / 2;
    grid-column: 1 / 2;
  }

  .charity-info {
    grid-row: 1 / 2;
    grid-column: 2 / 3;

    .charity-name {
      font-weight: 600;
      cursor: pointer;
    }
    .charity-publication-date {
      font-size: 13px;
      color: #707070;
    }
  }

  .follow {
    grid-row: 1 / 2;
    grid-column: 3 / 4;
  }

  .charity-description {
    max-width: 615px;
    grid-row: 2 / 3;
    grid-column: 2 / 3;

    button {
      margin: 8px 0;
    }
  }
}