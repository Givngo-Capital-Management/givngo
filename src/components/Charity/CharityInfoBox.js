import React from 'react';
import { Image, Button } from 'semantic-ui-react';
import styles from './CharityInfoBox.module.scss';

export default function CharityInfoBox(props) {
  const { charity } = props
  return (
    <div className={styles.box}>
      <Image
        className={styles.image}
        src={charity.logo}
        circular
      />
      <div className={styles.info}>
        <div className="charity-name">{charity.name}</div>
        <span>80k followers</span>
      </div>
      <Button className={styles.follow} color="orange">
        Follow
      </Button>
      <div className={styles.description}>
        <p>{charity.description}</p>
      </div>
    </div>
  );
}
