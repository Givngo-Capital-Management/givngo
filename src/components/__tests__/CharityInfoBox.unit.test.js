import React from 'react';
import { shallow } from 'enzyme';
import CharityInfoBox from '../Charity/CharityInfoBox';

describe('CharityInfoBox', () => {
  test('renders', () => {
    const wrapper = shallow(<CharityInfoBox />);
    expect(wrapper).toMatchSnapshot();
  });
});
