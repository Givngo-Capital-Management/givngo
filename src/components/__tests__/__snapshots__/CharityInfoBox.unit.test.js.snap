// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharityInfoBox renders 1`] = `
<div
  className="box"
>
  <Image
    as="img"
    circular={true}
    className="image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="info"
  >
    <div
      className="charity-name"
    >
      Charity Name
    </div>
    <span>
      80k followers
    </span>
  </div>
  <Button
    as="button"
    className="follow"
    color="orange"
  >
    Follow
  </Button>
  <div
    className="description"
  >
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </p>
  </div>
</div>
`;
